src/python/pants/help/help_info_extracter.py

```diff
@@ -781,6 +801,86 @@ def load() -> PluginAPITypeInfo:
             }
         )
 
+    @classmethod
+    def get_backend_help_info(cls, options: Options) -> LazyFrozenDict[str, BackendHelpInfo]:
+        DiscoveredBackend = namedtuple(
+            "DiscoveredBackend", "provider, name, register_py, enabled", defaults=(False,)
+        )
+
+        def discover_source_backends(root: Path, is_source_root: bool) -> set[DiscoveredBackend]:
+            provider = root.name
+            source_root = f"{root}/" if is_source_root else f"{root.parent}/"
+            register_pys = root.glob("**/register.py")
+            backends = {
+                DiscoveredBackend(
+                    provider,
+                    str(register_py.parent).replace(source_root, "").replace("/", "."),
+                    str(register_py),
+                )
+                for register_py in register_pys
+            }
+            return backends
+
+        def discover_plugin_backends(entry_point_name: str) -> set[DiscoveredBackend]:
+            backends = {
+                DiscoveredBackend(
+                    entry_point.dist.project_name,
+                    entry_point.module_name,
+                    str(
+                        Path(entry_point.dist.location)
+                        / (entry_point.module_name.replace(".", "/") + ".py")
+                    ),
+                    True,
+                )
+                for entry_point in pkg_resources.iter_entry_points(entry_point_name)
+                if entry_point.dist is not None
+            }
+            return backends
+
+        global_options = options.for_global_scope()
+        builtin_backends = discover_source_backends(
+            Path(pants.backend.__file__).parent.parent, is_source_root=False
+        )
+        inrepo_backends = (
+            chain.from_iterable(
+                discover_source_backends(Path(path_entry), is_source_root=True)
+                for path_entry in global_options.pythonpath
+            )
+            if "pythonpath" in global_options
```

Comment:
when would this be false?

Reply:
I think it was false if pythonpath was not set in `pants.toml`. Can double check.

Reply:
hmm, nope. that doesn't seem like it, so I'll drop that weird if else, thanks :)

Reply:
Actually, it's from the tests, that likely provides a too slim `global_options` value, will fix in the test instead, as this is not an issue when the value is not set in `pants.toml`.
```
>       all_help_info = HelpInfoExtracter.get_all_help_info(
            options,
            UnionMembership({}),
            fake_consumed_scopes_mapper,
            RegisteredTargetTypes({BazLibrary.alias: BazLibrary}),
            bc_builder.create(),
        )

src/python/pants/help/help_info_extracter_test.py:288:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/python/pants/help/help_info_extracter.py:546: in get_all_help_info
    name_to_backend_help_info=cls.get_backend_help_info(options),
src/python/pants/help/help_info_extracter.py:846: in get_backend_help_info
    for path_entry in global_options.pythonpath
src/python/pants/option/option_value_container.py:142: in __getattr__
    return self._get_underlying_value(key)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = OptionValueContainer(_value_map={'opt1': RankedValue(rank=<Rank.HARDCODED: 'HARDCODED'>, value=42, details=None), 'level': RankedValue(rank=<Rank.HARDCODED: 'HARDCODED'>, value=<LogLevel.INFO: 'info'>, details=None)}), key = 'pythonpath'

    def _get_underlying_value(self, key: Key):
        # Note that the key may exist with a value of None, so we can't just
        # test self._value_map.get() for None.
        if key not in self._value_map:
>           raise AttributeError(key)
E           AttributeError: pythonpath

src/python/pants/option/option_value_container.py:127: AttributeError
------------------------------------------- generated xml file: /private/var/folders/y2/6t5bbk554wv5h6w_qlgpcpnm0000gr/T/pants-sandbox-0xDOTq/src.python.pants.help.help_info_extracter_test.py.tests.xml -------------------------------------------
============================================================================================================== short test summary info ==============================================================================================================
FAILED src/python/pants/help/help_info_extracter_test.py::test_get_all_help_info - AttributeError: pythonpath
=========================================================================================================== 1 failed, 14 passed in 1.69s ============================================================================================================
```

Reply:
Fixed in the tests.